src/data/set/image.lean

```diff
@@ -471,6 +471,43 @@ end
 
 end image
 
+/-! ### Lemmas about the powerset and image. -/
+
+section powerset
+
+lemma mem_powerset_insert_iff {s t : set α} {x : α} :
+  s ∈ 𝒫 (insert x t) ↔ s ∈ 𝒫 t ∨ s ∈ (𝒫 t).image (insert x) :=
+begin
+  simp_rw [mem_powerset_iff],
+  constructor,
+  { intro h,
+    by_cases hs : x ∈ s,
+    { right,
+      use s \ {x},
+      constructor,
+      { rw [mem_powerset_iff, diff_singleton_subset_iff],
+        assumption },
+      { rw [insert_diff_singleton, insert_eq_of_mem hs] }},
+    { left,
+      exact (subset_insert_iff_of_not_mem hs).mp h}},
+  { intro h,
+    rcases h with h | ⟨s', h₁, h₂⟩,
+    { exact subset_trans h (subset_insert x t) },
+    { rw [←h₂],
+      exact insert_subset_insert h₁ }}
+end
+
+lemma mem_powerset_insert_iff' {s t : set α} {a : α}  :
```

Comment:
I don't think this lemma belongs in this file; it doesn't mention `image` anywhere.

Reply:
You're probably right, but I am also not a fan of having the two in separate files. I removed it for now and will consider suggesting a file `Mathlib.Data.Set.Powerset` once the port is complete.